### PR TITLE
Implementing evaluation for multible metrics.

### PR DESCRIPTION
**Pull request by MarvinTeichmann**
The framework now support evaluation on multible metrics. To do so, objective.evaluation() returns a list of eval tubles. Each eval tuble constist of a string name (e.g. "Accuracy") and a tensorflow operation computing the metric.

### CODE REVIEW

File: bin/tv-train [hunk 1]
```diff
@@ -8,15 +8,17 @@ from __future__ import division
 from __future__ import print_function
 
 import imp
+import pdb
```
Comment (MartinThoma) [inline bin/tv-train:11]: Why do you import this? (Seems not to be necessary for the code; only for debugging)

File: bin/tv-train [hunk 2]
```diff
@@ -8,15 +8,17 @@ from __future__ import division
...
 import logging
+import numpy as np
 import os.path
 import sys
 
 # configure logging
 if 'TV_IS_DEV' in os.environ and os.environ['TV_IS_DEV']:
     logging.basicConfig(format='%(asctime)s %(levelname)s %(message)s',
-                        level=logging.DEBUG,
+                        level=logging.INFO,
```
Comment (MartinThoma) [inline bin/tv-train:21]: Why did you change that? The reason for TV_IS_DEV is to have a DEBUG level output.

File: bin/tv-train [hunk 3]
```diff
@@ -136,14 +138,14 @@ def write_precision_to_summary(precision, summary_writer, name, global_step,
...
-def do_eval(hypes, eval_correct, phase, sess):
+def do_eval(hypes, eval_list, phase, sess):
     """Run one evaluation against the full epoch of data.
 
     Parameters
     ----------
     hypes : dict
         Hyperparameters
-    eval_correct : TODO
+    eval_dict : TODO
```
Comment (MartinThoma) [inline bin/tv-train:148]: Not correct anymore

Comment (MartinThoma) [inline bin/tv-train:148]: Also you should explain how it is used. An example might be nice.

File: bin/tv-train [hunk 4]
```diff
@@ -293,24 +306,26 @@ def run_training(hypes):
                 start_time = time.time()
 
             # Save a checkpoint and evaluate the model periodically.
-            if (step + 1) % 1000 == 0 or (step + 1) == solver['max_steps']:
+            if (step + 1) % int(utils.cfg.step_eval) == 0 or \
```
Comment (MartinThoma) [inline bin/tv-train:309]: We should probably do the int conversion in utils.

File: tensorvision/utils.py [hunk 5]
```diff
@@ -112,7 +112,7 @@ def _set_cfg_value(cfg_name, env_name, default, cfg):
                os.path.expanduser("~/tv-plugins"),
                cfg)
 _set_cfg_value('step_show', 'TV_STEP_SHOW', 100, cfg)
-_set_cfg_value('step_eval', 'TV_STEP_EVAL', 100, cfg)
+_set_cfg_value('step_eval', 'TV_STEP_EVAL', 200, cfg)
```
Comment (MartinThoma) [inline tensorvision/utils.py:115]: Is there a good reason to change this value?

**Comment by MarvinTeichmann** [comment]
Merged in 25b2f634550d12987f247ac1d0c5db6824bb8c81
